client/src/game/ui/tools/select.vue

```diff
@@ -190,7 +192,11 @@ export default class SelectTool extends Tool {
         const layer = layerManager.getLayer()!;
 
         if (this.mode === SelectOperations.GroupSelect) {
-            layer.clearSelection();
+            if (keyboardState.shiftPressed || keyboardState.ctrlPressed) {
+                // If either control or shift are pressed, do not remove selection
```

Comment:
If you draw a new box in shiftmode that also contains a shape that already was in a selection, it now is included twice in the selection.  This needs to be handled in line 211!

Reply:
I did not notice this. I believe this would be especially problematic when pasting shapes. I will add a check to not add shapes to the selection multiple times.